src/components/Community/CommunityHolderList.tsx

```diff
@@ -0,0 +1,116 @@
+import Loader from 'components/core/Loader/Loader';
+import { VStack } from 'components/core/Spacer/Stack';
+import TokenHolderList from 'components/TokenHolderList/TokenHolderList';
+import { LIST_ITEM_PER_PAGE } from 'constants/community';
+import { GLOBAL_FOOTER_HEIGHT } from 'contexts/globalLayout/GlobalFooter/GlobalFooter';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { graphql, usePaginationFragment } from 'react-relay';
+import styled from 'styled-components';
+import { CommunityHolderListFragment$key } from '__generated__/CommunityHolderListFragment.graphql';
+
+type Props = {
+  communityRef: CommunityHolderListFragment$key;
+};
+
+export default function CommunityHolderList({ communityRef }: Props) {
+  const { data: community, loadNext, hasNext } = usePaginationFragment(
+    graphql`
+      fragment CommunityHolderListFragment on Community
+        @refetchable(queryName: "CommunityHolderListRefetchableFragment") {
+        id
+
+        owners(
+          first: $listOwnersFirst
+          after: $listOwnersAfter
+          onlyGalleryUsers: $onlyGalleryUsers
+        ) @connection(key: "CommunityPageView_owners") {
+          edges {
+            node {
+              __typename
+
+              user {
+                universal
+              }
+
+              ...TokenHolderListFragment
+            }
+          }
+        }
+      }
+    `,
+    communityRef
+  );
+  const [isFetching, setIsFetching] = useState(false);
+
+  const handleSeeMore = useCallback(() => {
+    setIsFetching(true);
+    loadNext(LIST_ITEM_PER_PAGE, {
+      onComplete: () => {
+        setIsFetching(false);
+      },
+    });
+  }, [loadNext]);
+
+  useEffect(() => {
+    function handleScrollPosition() {
+      const pixelsFromBottomOfPage =
+        document.body.offsetHeight - window.pageYOffset - window.innerHeight;
+      if (pixelsFromBottomOfPage < GLOBAL_FOOTER_HEIGHT && hasNext && !isFetching) {
+        handleSeeMore();
+      }
+    }
+
+    window.addEventListener('scroll', handleScrollPosition);
+
+    return () => window.removeEventListener('scroll', handleScrollPosition);
+  }, [handleSeeMore, hasNext, isFetching]);
```

Comment:
my gut tells me that this would add a ton of scroll listeners as the dependencies update (`isFetching`, etc.)

however, when i look at the console i see that we're only making 1 network request when we try to scroll/paginate, which is good...

@tbezman do you think this is OK?

Reply:
@Robinnnnn as `isFetching` changes, the previous handler0s will get cleaned up. and if for some reason, `handleSeeMore` gets doubly triggered, Relay should end up deduping those requests anyway

Reply:
@tbezman I see. So, should we let it be like this or what's your advice/recommendation solution on this?